syntaxes/test/data/expression.html

```diff
@@ -0,0 +1,66 @@
+<!-- Interpolation test -->
```

Comment:
Let's just put each test case in an interpolation like `{{ }}`, each on a separate line and testing a clear, individual thing. Some "integration" tests for combinations of grammar scopes can be included at the end too. But the tags and attributes are fluff that we test in other files, and I think here it is much more important to test that we handle operators, variable assignments, pipes, etc. correctly.

If possible, can you include one test case for every scope you define in your grammar and some integrations between them at the end? I realize this is a big task; if you would like some help, please let me know and I will work on it as well.

Reply:
Sorry for taking so long to answer. I had to work off shift yesterday.

I think the best option for now would be covering the main expression cases (all of the common expression cases). All of the scopes wont be possible now as there are some unused scopes still present in the code. After having this PR merged, I'll focus on reducing a bit more the grammar to clear up those unused scopes.

Does that sounds good?

BTW: When working on reducing this grammar, I'll eventually be adding these changes to the grammar in my extension as well. Would you mind if I "borrow" the way this extension approaches snapshot tests? I find it very productive.

Reply:
I'll be adding test cases to the html file and as soon as you find it to be enough, I'll rebase on master and I think we're ready to go.

Reply:
Sure, that all sounds good. Feel free to reuse as much of anything as you would like.

Following this PR we should talk about a formal grammar for expressions. There are actually two things going on here: a grammar for expressions as they exist e.g. inside interpolations, and the microsyntax that we have for template bindings. I'll present a doc later on.

Reply:
That's okay. We can do that. 

Sorry for the long time to asnwer. i've been busier than I expected

Reply:
No worries. I will work on this today.